File: matiasmasso/Mat.NET/DTO/DTOBaseQuota.cs
using System;

namespace DTO
{
    public class DTOBaseQuota
    {
        public object source { get; set; }
        public DTOAmt baseImponible { get; set; }
        public DTOAmt quota { get; set; }
        public decimal tipus { get; set; }

        public DTOBaseQuota(DTOAmt oBase, decimal DcTipus = 0, DTOAmt oQuota = null/* TODO Change to default(_) if this is not a reference type */) : base()
        {
            baseImponible = oBase;
            if (DcTipus != 0)
            {
                tipus = DcTipus;
                if (oQuota == null)
                    quota = oBase.Percent(DcTipus);
                else
                    quota = oQuota;
            }
        }

        public void calcTipus()
        {
            if (baseImponible == null)
                tipus = 0;
            else if (baseImponible.Eur == 0)
                tipus = 0;
            else
                tipus = 100 * Math.Round(quota.Eur / baseImponible.Eur, baseImponible.Cur.Decimals, MidpointRounding.AwayFromZero);
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOBaseQuota.vb
Public Class DTOBaseQuota
    Property Source As Object
    Property Base As DTOAmt
    Property Quota As DTOAmt
    Property Tipus As Decimal

    Public Sub New(oBase As DTOAmt, Optional DcTipus As Decimal = 0, Optional oQuota As DTOAmt = Nothing)
        MyBase.New
        _Base = oBase
        If DcTipus <> 0 Then
            _Tipus = DcTipus
            If oQuota Is Nothing Then
                _Quota = oBase.Percent(DcTipus)
            Else
                _Quota = oQuota
            End If
        End If
    End Sub

    Public Sub CalcTipus()
        If _Base Is Nothing Then
            _Tipus = 0
        ElseIf _base.Eur = 0 Then
            _Tipus = 0
        Else
            _Tipus = 100 * Math.Round(_Quota.Eur / _Base.Eur, _Base.Cur.Decimals, MidpointRounding.AwayFromZero)
        End If
    End Sub
End Class
